safe-global/safe-wallet-web
# Feat: Scan EIP-712 typed data signature requests

## What it solves

Partially resolves #2113 

## How this PR fixes it
- Adds risk analysis to EIP-712 signature requests via Redefine
- Reuses the same components to display results / warnings

## How to test it
- Sign any EIP-712 message from a Safe
- Observe "Scan for risks"-block
- Ideally also try some malicious signature requests

## Screenshots
![Screenshot 2024-01-03 at 11 14 30](https://github.com/safe-global/safe-wallet-web/assets/2670790/2beb9f0b-691b-4744-8e6c-457646c7704b)

## Checklist
